### PR TITLE
feat(ɵtoolkit): allow organizing content in tabs

### LINKED ISSUE #10: TabBar widget for DevTools
## Is your feature request related to a problem? Please describe.
The DevTools that are being developed for the SCION Microfrontend Platform (see https://github.com/SchweizerischeBundesbahnen/scion-microfrontend-platform/issues/4) will provide a lot of information about the micro applications and their capabilities. Since the DevTools will be displayed as part of the application (similar to the developer tools in Chrome), the available space is limited. Therefore all the information has to be presented in a compact manner.

## Describe the solution you'd like
In order to be able to split the information and present only part of it, a tab bar is a natural choice. The user can then switch between tabs to display the corresponding piece of information.

### PR DESCRIPTION
**Pull request by mofogasy**
closes #10

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-toolkit/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Fix
- [x] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, updating dependencies, removal of deprecations, etc)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number: #10 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/scion/toolkit.internal/widgets/src/tabbar/public_api.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ */
+
+export { SciTabbarModule } from './tabbar.module';
+export { SciTabbarComponent } from './tabbar.component';
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/public_api.ts:12]: You also need to export `SciTabDirective`.

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ * ---
+ * Example usage:
+ *
+ * <sci-tabbar>
+ *   <ng-template sciTab [label]="contact.name" *ngFor="let contact of contacts$ | async">
+ *     <app-contact-list-item [contact]="contact"></app-contact-list-item>
+ *   </ng-template>
+ * </sci-tabbar>
+ */
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:27]: Please use a neutral example, as in `SciTabDirective`.
```
<sci-tabbar>
  <ng-template sciTab [label]="item.label" *ngFor="let item of items$ | async">
   ...
  </ng-template>
</sci-tabbar>
```

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+   */
+  public detachContent(): void {
+    if (this.isContentAttached()) {
+      this._vcr.detach(this._vcr.indexOf(this._viewRef));
+      this._viewRef.detach();
+    }
+  }
+
+  /**
+   * Returns if this tab's content is currently attached, i.e., visible to the user.
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:75]: Please change as following: `Returns whether the tab content is currently attached to the DOM, meaning that the user has selected the tab.`

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+   */
+  public isContentAttached(): boolean {
+    return !!this._vcr && this._vcr?.indexOf(this._viewRef) !== -1;
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:78]: If using the safe navigation operator, you can simplify it as following:
```
return this._vcr?.indexOf(this._viewRef) !== -1 || false;
```

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      margin: 0;
+      padding: 0;
+
+      > li.tab, > li.filler {
+        flex: none;
+        border-bottom: 3px solid var(--sci-color-P300);
+        padding: 0 1em 0 1em;
+        white-space: nowrap;
+
+        &:hover {
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.scss:22]: Also exclude the filler, as following:
```&:hover:not(.filler)```
or combine it with the subsequent style, as following:
```
&:hover:not(.filler) {
  cursor: pointer;

  &:not(.selected) {
    border-bottom-color: var(--sci-color-P400);
    color: var(--sci-color-accent);
  }
}
```

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  public onTabSelect(tab: SciTabDirective | undefined) {
+    this.selectTab(tab);
+  }
+
+  private selectTab(tab: SciTabDirective | undefined) {
+    // Deactivate the currently selected tab, if any.
+    const selectedTab = this.tabs.find(it => it.isContentAttached());
+    if (selectedTab) {
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:72]: Do nothing when clicking on the already active tab, as follows:

```
const selectedTab = this.tabs.find(it => it.isContentAttached());
if (selectedTab && selectedTab === tab) {
   return;
}
```

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  private onTabsChange(tabs: QueryList<SciTabDirective>): void {
+    // Select the first tab if no tab is selected or if the active tab was removed
+    if (!this._activeTab || !tabs.find(it => it === this._activeTab)) {
+      this.selectTab(tabs.first);
+    }
+  }
+
+  public onTabSelect(tab: SciTabDirective | undefined) {
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:65]: Also, specify the return type, as follows:
```public onTabSelect(tab: SciTabDirective | undefined): void {```

Could you please check why this does not lead to a lint error?

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    this.selectTab(tab);
+  }
+
+  private selectTab(tab: SciTabDirective | undefined) {
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:69]: Same here with the missing return type.

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    // Deactivate the currently selected tab, if any.
+    const selectedTab = this.tabs.find(it => it.isContentAttached());
+    if (selectedTab) {
+      selectedTab.detachContent();
+    }
+
+    // Activate the new tab, if any.
+    if (tab) {
+      this._activeTab = tab;
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:78]: I suggest not to introduce state with the currently active tab, but instead, add a method to compute the currently active tab, as follows:
```
private getActiveTab(): SciTabDirective | undefined {
  return this.tabs.find(tab => tab === tab.isContentAttached())
}
```
You can then use this method on lines `60` and `71` .

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+  public ngAfterContentInit(): void {
+    this.tabs$ = this.tabs.changes
+      .pipe(
+        startWith(this.tabs),
+        tap(tabs => this.onTabsChange(tabs))
+      );
+  }
+
+  private onTabsChange(tabs: QueryList<SciTabDirective>): void {
```
Comment (danielwiehl) [inline projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:58]: Name the method according to how it works. Then you can omit your comment.

For example: ```ensureTabActivated```, or ```activateFirstTabIfNoneActivated```, or ...

You may not even need a method if you add a `getActiveTab` method:
``` tap(tabs => this.getActiveTab() ? noop() : this.selectTab(tabs.first))```

File: projects/scion/toolkit.internal/widgets/src/tabbar/public_api.ts (hunk 1 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/public_api.ts:12]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts (hunk 2 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:27]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts (hunk 3 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:75]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts (hunk 4 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts:78]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.scss (hunk 5 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.scss:22]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts (hunk 6 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:72]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts (hunk 7 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:65]: Done. The linting problem has been identified and will be addressed in another issue.

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts (hunk 8 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:69]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts (hunk 9 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:78]: done

File: projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts (hunk 10 above)
Comment (mofogasy) [reply projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts:58]: done

**Pull request merged**